### ISSUE TITLE
[webview_flutter] Remove manual registration

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#117482]
Currently we document the need to [manually register `webview_flutter_web`](https://pub.dev/packages/webview_flutter_web#registering-the-implementation). IIRC this was only necessary because of the broken federation structure of `webview_flutter`, which v4 fixed.

@bparrishMines We can just add the registration to `registerWith(...)` now, right?

**Comment by ariadp2022** [comment]
Hi, webview_flutter_web is not an endorsed implementation of the webview_flutter plugin yet. 
I can not update the webview_flutter to 4.0.0. I'm need the web platform implementaion in my application for pwa.

**Comment by stuartmorgan** [comment]
This issue isn't about endorsing the current implementation, it's about making it easier to use while unendorsed.

**Comment by bparrishMines** [comment]
Actually, it looks like the latest version of `webview_flutter_web` was never published: https://github.com/flutter/plugins/commit/ca974ab0c37c3cf11c650eecec1142dcbc690bab.

One of the `ios_platform_tests` never finished and the release action never went through. 

Should we go ahead and do a manual publish of the plugin? This also seems to indicate a bug in our testing infrastructure.

**Comment by ariadp2022** [comment]
Because no versions of webview_flutter_web match >0.1.0+4 <0.2.0 and webview_flutter_web 0.1.0+4 depends on webview_flutter_platform_interface ^1.8.0, webview_flutter_web ^0.1.0+4 requires webview_flutter_platform_interface ^1.8.0.
And because webview_flutter >=4.0.0 depends on webview_flutter_platform_interface ^2.0.0, webview_flutter_web ^0.1.0+4 is incompatible with webview_flutter >=4.0.0.

**Comment by ekuleshov** [comment]
> when update webview_flutter package get this error
> 
> Because no versions of webview_flutter_web match >0.1.0+4 <0.2.0 and webview_flutter_web 0.1.0+4 depends on webview_flutter_platform_interface ^1.8.0, webview_flutter_web ^0.1.0+4 requires webview_flutter_platform_interface ^1.8.0. And because webview_flutter >=4.0.0 depends on webview_flutter_platform_interface ^2.0.0, webview_flutter_web ^0.1.0+4 is incompatible with webview_flutter >=4.0.0.

Once updated `webview_flutter_web` is published you will be able to update to `0.2.0` version.

In the mean time the following dependency declaration worked for me:

```
  webview_flutter_web:
    git:
      url: https://github.com/flutter/plugins.git
      path: packages/webview_flutter/webview_flutter_web
      ref: 9dc05342850b123f0827f159beaa2568adfff2b9 # or ca974ab0c37c3cf11c650eecec1142dcbc690bab
```

**Comment by ariadp2022** [comment]
setjavaScriptMode, setBackgroundColors, setNavigationDelegat methods are not implement on this plugin for platform web
